clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/ops/DeployAppengineAtomicOperation.groovy

```diff
@@ -57,10 +55,26 @@ class DeployAppengineAtomicOperation implements AtomicOperation<DeploymentResult
 
   DeployAppengineAtomicOperation(DeployAppengineDescription description) {
     this.description = description
-    this.containerDeployment = description.containerImageUrl?.trim()
-    this.usesGcs = !this.containerDeployment && description.repositoryUrl.startsWith("gs://")
+    if (description.artifact) {
+      switch(description.artifact.type) {
+        case 'gcs/object':
+          if (!description.artifact.reference) {
+            throw new AppengineOperationException("Missing artifact reference for GCS deploy")
+          }
+          usesGcs = true;
+        case 'docker/image':
+          if (!description.artifact.name) {
```

Comment:
It's interesting that these are using different fields to fetch from -- as the list of supported artifacts expands (not just in this case, but overall) how are we planning to document and keep track of fetching/writing each type? More of a general question not specific to this PR necessarily. @lwander thoughts?

Reply:
See my comment above - but the gist is that the artifact used for the app.yml should be passed to some artifact downloader (which exists in clouddriver-artifacts) that accepts any artifact that can be read from remote storage, and the artifact for the docker image should have its type enforced.